### PR TITLE
Fix SessionManager and account switches

### PR DESCRIPTION
**Pull request by tiagonapoli**
#### What is the purpose of this pull request?
- Use only SessionManager for managing session.
- Remove session related function on conf.
- Create unified function for switching back to previous account
- Remove unnused code
- Refactor `vtex login`
- Do not report `CommandError`.

#### How should this be manually tested?
```
git checkout fix/session-management

yarn watch
```
- Check if `vtex-test login` is working properly - try logging in to an account and check `~/.vtex/session/session.json` and `~/.vtex/session/workspace.json`.
- Check if `vtex-test switch` is working properly
- Try `vtex-test deprecate` an innexistent app to check switching accounts (`vtex-test deprecate basedevmkp.asdf@0.0.1`
- Try `vtex-test undeprecate` an innexistent app to check switching accounts (`vtex-test undeprecate basedevmkp.asdf@0.0.1`
- Try `vtex-test publish` an app with typescript errors to check switching accounts

#### Types of changes
- [X] Refactor (non-breaking change that only makes the code better)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

#### Chores checklist
- [X] Update `CHANGELOG.md`

### CODE REVIEW

File: src/lib/session/SessionManager.ts [hunk 1]
```diff
@@ -94,29 +120,74 @@ export class SessionManager {
...
+      this.state.lastAccount = this.account
+      this.state.lastWorkspace = null
+    }
+
+    const cachedToken = new Token(this.sessionPersister.getAccountToken(newAccount))
+    if (useCachedToken && cachedToken.isValid()) {
+      this.state.account = newAccount
+      this.state.workspace = targetWorkspace
+      this.state.tokenObj = cachedToken
+      this.flushState()
```
Comment (Jeymisson) [inline src/lib/session/SessionManager.ts:133]: This name `flushState` gives the impression we are disposing the state, but we are actually refreshing it.

Comment (tiagonapoli) [reply src/lib/session/SessionManager.ts:133]: How about `saveState`? The idea of `flushState` was to send it to the underlying sessionPersister implementation to save it to disk, for example

Comment (Jeymisson) [reply src/lib/session/SessionManager.ts:133]: I think this one works better. `saveState` or `updateState`.

File: src/modules/support/login.ts [hunk 2]
```diff
@@ -60,17 +62,18 @@ const assertToken = (raw: string): void => {
 }
 
 const saveSupportCredentials = (account: string, token: string): void => {
-  saveAccount(account)
-  saveWorkspace('master')
-  saveToken(token)
+  const session = SessionManager.getSingleton()
+  session.DEPRECATEDchangeAccount(account)
```
Comment (Jeymisson) [inline src/modules/support/login.ts:66]: Do you really want the deprecated one?

Comment (tiagonapoli) [reply src/modules/support/login.ts:66]: In this case I don't really know how `support login` works, nor how to test it - I created the deprecated one just for this. In a future PR we would want to change it

Comment (tiagonapoli) [reply src/modules/support/login.ts:66]: This should use `session.login`, providing a login method, implemented here:
https://github.com/vtex/toolbelt/blob/e3abd57ab55eb80d648f7f53d20e1c6054090b6a/src/lib/auth/AuthProviders/AuthProviders.ts#L5-L9

**Pull request merged**